compiler/src/vm/fiber.rs

```diff
@@ -500,6 +475,47 @@ impl Fiber {
             }
         }
     }
+
+    pub fn call(&mut self, callee: Pointer, arguments: Vec<Pointer>, responsible: Pointer) {
+        match &self.heap.get(callee).data {
+            Data::Closure(Closure {
+                captured,
+                num_args: expected_num_args,
+                ..
+            }) => {
+                if arguments.len() != *expected_num_args {
+                    self.panic(
+                        format!("A closure expected {expected_num_args} parameters, but you called it with {} arguments.", arguments.len()),
+                        self.heap.get_hir_id(responsible),
+                    );
+                    return;
+                }
+
+                self.call_stack.push(self.next_instruction);
+                self.data_stack.append(&mut captured.clone());
+                for captured in captured.clone() {
+                    self.heap.dup(captured);
+                }
```

Comment:
`vec::append(…)` empties the other `Vec`: https://doc.rust-lang.org/std/vec/struct.Vec.html#method.append. Hence, this would not `dup(…)` any objects right now

```suggestion
                for captured in captured.clone() {
                    self.heap.dup(captured);
                }
                self.data_stack.append(&mut captured.clone());
```